### PR TITLE
Update some comments on public APIs and types

### PR DESCRIPTION
**Pull request by bob-wilson**
This cleans up and clarifies some comments that I came across while reviewing the libra-vm validation code.

## Motivation

Improving documentation in the code

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Ran existing tests.

### CODE REVIEW

File: types/src/transaction/mod.rs [hunk 1]
```diff
@@ -633,8 +642,17 @@ impl GovernanceRole {
...
 pub struct VMValidatorResult {
+    /// Result of the validation: `None` if the transaction was successfully validated
+    /// or `Some(DiscardedVMStatus)` if the transaction should be discarded.
     status: Option<DiscardedVMStatus>,
+
+    /// Score for ranking the transaction priority (e.g., based on the gas price).
+    /// Only used when the status is `None`. Higher values indicate a higher priority.
     score: u64,
+
+    /// The account role for the transaction sender, so that the certain
```
Comment (dariorussi) [inline types/src/transaction/mod.rs:653]: remove "the"? "so that certain..."

**Comment by bob-wilson** [comment]
> Overall I think we should stay away from Rust comments `///` in favor of normal comments `//` but I see this was more to make things more or less consistent....

See the "Code documentation" section of documentation/coding_guidelines.md: "Any public fields, functions, and methods should be documented with Rustdoc". I'm not sure we're following that style very well, but we should at least work toward that direction.

**Pull request closed without merging**